crates/gpui/src/platform/windows/directx_renderer.rs

```diff
@@ -617,6 +630,52 @@ impl DirectXRenderer {
             driver_info: driver_version,
         })
     }
+
+    pub(crate) fn get_font_info() -> &'static FontInfo {
+        static CACHED_FONT_INFO: OnceLock<FontInfo> = OnceLock::new();
+        CACHED_FONT_INFO.get_or_init(|| unsafe {
+            let factory: IDWriteFactory5 = DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED).unwrap();
+            let render_params: IDWriteRenderingParams1 =
+                factory.CreateRenderingParams().unwrap().cast().unwrap();
+            FontInfo {
+                gamma_ratios: Self::get_gamma_ratios(render_params.GetGamma()),
+                grayscale_enhanced_contrast: render_params.GetGrayscaleEnhancedContrast(),
+            }
+        })
+    }
+
+    // Gamma ratios for brightening/darkening edges for better contrast
+    // https://github.com/microsoft/terminal/blob/1283c0f5b99a2961673249fa77c6b986efb5086c/src/renderer/atlas/dwrite.cpp#L50
```

Comment:
I'm super happy to see my DWrite HLSL shader ending up in other big projects. In fact, this is the first other, big project that I'm aware of. 🙈

While your HLSL code has a different structure, it is the same fundamentally, however. Because of this, I would suggest adding (for instance) the Terminal license notice verbatim in both places and a link back, like this line already has it.

(I'm not speaking as a MSFT employee, but rather as an individual. I simply consider proper attribution to be important, no matter whether it's StackOverflow or similar.)

Edit: Apropos, I think I'm effectively "the text rendering person" at Microsoft right now. So, if you ever have any questions, let me know / ping me. 🙂

Reply:
I'm genuinely so happy to hear this! Please don't hesitate to reach out to me if you ever need anything. 😊

FYI, you can find the related shader code here: https://github.com/microsoft/terminal/blob/1283c0f5b99a2961673249fa77c6b986efb5086c/src/renderer/atlas/dwrite.hlsl